src/Recommender/Engagement/ImageTracked.jsx

```diff
@@ -0,0 +1,31 @@
+const handleClick = () => {
+  State.update({ clicked: true });
+};
+
+State.init({
+  clicked: false,
+});
+
+return (
+  <div onClick={handleClick}>
+    <Widget
+      src="${REPL_ACCOUNT}/widget/Recommender.Service.EngagementTracker"
```

Comment:
@scopalaffairs Did you explore using `Vm.Require` introduced by this  [PR on the VM?](https://github.com/NearSocial/VM/pull/61)

Just curious.

Reply:
@roshaans, I did take a look at it and find it an appealing concept, but did not test it, and since wrapping the target components in the service already worked as expected, I decided to stick with a working solution.